src/Tokenizer/TokensAnalyzer.php

```diff
@@ -633,6 +633,28 @@ public function isWhilePartOfDoWhile(int $index): bool
         return $tokens[$beforeStartIndex]->isGivenKind(T_DO);
     }
 
+    public function isEnumCase(int $caseIndex): bool
+    {
+        $tokens = $this->tokens;
+        $token = $tokens[$caseIndex];
+
+        if (!$token->isGivenKind(T_CASE)) {
+            throw new \LogicException(sprintf(
+                'No T_CASE given at index %d, got %s instead.',
+                $caseIndex,
+                $token->getName() ?? $token->getContent()
+            ));
+        }
+
+        if (!\defined('T_ENUM') || !$tokens->isTokenKindFound(T_ENUM)) {
+            return false;
+        }
+
+        $prevIndex = $tokens->getPrevTokenOfKind($caseIndex, [[T_ENUM], [T_SWITCH]]);
+
+        return null !== $prevIndex && $tokens[$prevIndex]->isGivenKind(T_ENUM);
```

Comment:
I think that with this implementation non-enum cases inside enum (like in `pure enum with switch` test case) won't be considered as statements too, so they won't be fixed with `--rules='{"blank_line_before_statement":{"statements":["case"]}}'` configuration.

Reply:
switch cases inside enums will still be fixed. Look at the modified provideFix81Cases of BlankLineBeforeStatementFixerTest

Reply:
OK, I tested it wrong. I took `enum Suit` from and applied fixer, but it has two cases with same logic and new line is not applied then. When cases have different logic, new line is added:

<img width="1032" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/3314ddf9-e431-4d6e-9d01-900dc70d403b">

And yeah, now I get the logic here - find `enum` or `switch`, but only on of them makes it an "enum case" 👍.